src/VisualStudio/Core/Def/DocumentOutline/ILanguageServiceBrokerShim.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Threading;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageServiceBrokerShim
+{
+    // This shim type is required as ILanguageServiceBroker2 cannot be imported directly since the Microsoft.VisualStudio.LanguageServer.Client.Implementation
+    // package reference exists in Microsoft.VisualStudio.LanguageServices.Implementation (instead of Microsoft.VisualStudio.LanguageServices).
```

Comment:
I'm not quite sure I follow this explanation -- I think there's a backstory that's not being explained here?

Reply:
`Microsoft.VisualStudio.LanguageServer.Client.Implementation` does not ship on nuget but `Microsoft.VisualStudio.LanguageServices` does so we cannot depend on it directly we instead need this shim to act as a redirect (right @dibarbet ?)

@emilyanas2323 I would just update this with that explanation.

Reply:
Right!  For our public nuget packages their dependencies should also be on nuget.